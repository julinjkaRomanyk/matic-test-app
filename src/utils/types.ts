export type InfoStore = {
    firstName: string;
    lastName: string;
    phoneNumber: string;
    stateName: string;
};

export type Progress = {
    current: number;
    total: number;
};

export enum Fields {
    nameQuestion = "nameQuestion",
    phoneQuestion = "phoneQuestion",
    stateQuestion = "stateQuestion",
};


export type DefaultOption = { 
    label: string; 
    value: string 
};