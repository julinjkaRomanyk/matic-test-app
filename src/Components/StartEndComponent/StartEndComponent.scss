.Component-Wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    max-width: 70%;

    img {
        max-width: 40%;
        margin-top: 100px;
    }
}

.Component-Message {
    font-size: 48px;
    padding: 20px;
}

@media screen and (max-width: 1000px) {
    .Component-Message {
        font-size: 38px;
    }
}

@media screen and (max-width: 590px) {
    .Component-Message {
        font-size: 28px;
    }

    .Component-Wrapper img {
        max-width: 45%;
    }
}
