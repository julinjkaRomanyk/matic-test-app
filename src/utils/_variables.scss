$BasicBlack: black;
$BasicOrange: #FF5122;
$BasicWhite: white;
$SecondOrange: #FFBE1E;
$BackgroundColor: #FCFAF9;
$Grey: #726969;
$BorderColor: #AFA7A6;
$DarkPowder: #F9ECE7;
