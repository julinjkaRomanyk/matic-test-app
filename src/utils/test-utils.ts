export function getTestName(dirname: string = __dirname) {
    return dirname.split("/").reverse()[0];
  };