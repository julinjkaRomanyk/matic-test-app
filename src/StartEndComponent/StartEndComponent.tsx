import React from "react";

import { Button } from "./../Button";
import "./StartEndComponent.scss";

export type Props = {
    message: string;
    image: string;
    buttonLabel?: string;

    onButtonClick?(): void;

}

export const StartEndComponent: React.FC<Props> = ({ image, message, onButtonClick, buttonLabel = "Start" }) => {

    return (
        <div className="Component-Wrapper">
            <img src={image} alt="Matic" />
            <div className="Component-Message"> {message} </div>
            {!!onButtonClick && <Button label={buttonLabel} onClick={onButtonClick!} />}
        </div>
    )
}