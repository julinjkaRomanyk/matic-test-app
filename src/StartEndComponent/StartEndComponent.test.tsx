import React from "react";
import { getTestName } from "../utils/test-utils";
import { render, fireEvent } from "@testing-library/react";

import { StartEndComponent } from "./StartEndComponent";

const startComponentProps = {
message: "Hello",
buttonLabel: "Start",
image: "image",
onButtonClick: jest.fn()
}

describe(getTestName(__dirname), () => {
    test("Match snapshot", () => {
      const { container } = render(<StartEndComponent {...startComponentProps} />);
  
      expect(container.firstChild).toMatchSnapshot();
    });

    test("Button should be rendered in case Start Component", () => {
      const { getByTestId } = render(<StartEndComponent {...startComponentProps} />);

      const startButton = getByTestId("button");
  
      expect(startButton).toBeTruthy();
    });

    test("onButtonClick should be called in case button is clicked", () => {
      const { getByTestId } = render(<StartEndComponent {...startComponentProps} />);

      const startButton = getByTestId("button");

      fireEvent.click(startButton);
  
      expect(startComponentProps.onButtonClick).toHaveBeenCalled();
    });
  });
  
