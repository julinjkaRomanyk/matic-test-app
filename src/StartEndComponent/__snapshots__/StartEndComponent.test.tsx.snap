// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartEndComponent Match snapshot 1`] = `
<div
  class="Component-Wrapper"
>
  <img
    alt="Matic"
    src="image"
  />
  <div
    class="Component-Message"
  >
     
    Hello
     
  </div>
  <button
    class="Button"
    data-testid="button"
    id="button"
  >
    Start
  </button>
</div>
`;
